crates/api_crud/src/post/create.rs

```diff
@@ -91,6 +92,25 @@ impl PerformCrud for CreatePost {
     let (embed_title, embed_description, embed_video_url) = metadata_res
       .map(|u| (u.title, u.description, u.embed_video_url))
       .unwrap_or_default();
+    let language = data.language.clone();
+    let mut language = blocking(context.pool(), move |conn| {
+      Language::read_id_from_code_opt(conn, language)
+    })
+    .await??;
+    // if user only speaks one language, use that as post language. otherwise, set it as "undetermined"
+    if language.is_none() {
+      let user_langs = local_user_view.local_user.discussion_languages;
```

Comment:
Discussion languages isn't a column of local_user, its a new table that has to be added. So you'll need to do a query to fetch those discussion_languages.